lib/awful/ewmh.lua

```diff
@@ -355,11 +355,11 @@ function ewmh.merge_maximization(c, context, hints)
         c._delay_maximization = function()
             -- This ignores unlikely corner cases like mismatching toggles.
             -- That's likely to be an accident anyway.
-            if c._delayed_max_h and c._delayed_max_v then
+            if c._delayed_max_h ~= nil and c._delayed_max_v ~= nil then
```

Comment:
Umm, looking at this again, wont this set `c.maximized` to true even if one of them is false? That's not the intended behavior.

Reply:
Agreed. I don't have a perfect solution in mind. Before this patch it is not possible to set maximized to false in this if branch. Maybe changing the assignment to `c.maximized = ... and ...` would make more sense - what do you think?

Reply:
Maybe a variant of:

```
if c._delayed_max_h == true and c._delayed_max_v == true then
    c.maximized = true
elseif c._delayed_max_h == false and c._delayed_max_v == false then
    c.maximized = false
elseif ...
```

?

Because otherwise it cannot be merged. This cause a rather grave regression for every programs for the sake of fixing something that only happens with CSD apps, aka the < 100 Gnome3 apps that use it.

Reply:
I refactor the logic here to interpret the client's intention, based on the client's view of how the X11 atoms would change -- please let me know what do you think.